polars/polars-io/src/csv/write_impl.rs

```diff
@@ -267,7 +262,11 @@ pub(crate) fn write<W: Write>(
             let mut finished = false;
             // loop rows
             while !finished {
-                for col in &mut col_iters {
+                for (i, col) in &mut col_iters.iter_mut().enumerate() {
+                    let datetime_format = match &options.datetime_format {
+                        Some(datetime_format) => Some(datetime_format.as_str()),
+                        None => unsafe { *formats.get_unchecked(i % n_columns) },
```

Comment:
Modulo's are expensive. This modulo should not be needed as `cols_iter` returns an iterator per column and should be depleted once we have iterated all columns.